validation/custom_constraint.rst

```diff
@@ -69,6 +69,10 @@ The validator class is also simple, and only has one required method ``validate(
                 throw new UnexpectedTypeException($value, 'string');
             }
 
+            if ('' === $value) {
```

Comment:
`null` must be check too.

Reply:
But the `null` value would never be checked. We throw an exception if `!is_string()` above.

Reply:
This check must be moved before indeed, and test null.

Reply:
The reason is that the constraint should not throw on invalid empty values, or it would superseed other constraints responsibility.